src/utils/components/InstanceTypeModal/InstanceTypeModal.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { FC, useMemo, useState } from 'react';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Alert,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Flex,
+  FlexItem,
+  Modal,
+  SelectList,
+  SelectOption,
+  Text,
+} from '@patternfly/react-core';
+import { InstanceTypeUnion } from '@virtualmachines/details/tabs/configuration/utils/types';
+
+import FormPFSelect from '../FormPFSelect/FormPFSelect';
+
+import {
+  getInstanceTypeFromSeriesAndSize,
+  getInstanceTypeSeriesAndSize,
+  getInstanceTypeSeriesDisplayName,
+  getInstanceTypesPrettyDisplaySize,
+  getInstanceTypesSizes,
+  mappedInstanceTypesToSelectOptions,
+} from './utils/util';
+
+type InstanceTypeModalProps = {
+  allInstanceTypes: InstanceTypeUnion[];
+  instanceType: InstanceTypeUnion;
+  instanceTypeVM: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (
+    updatedVM: V1VirtualMachine,
+    instanceType: InstanceTypeUnion,
+  ) => Promise<V1VirtualMachine>;
+};
+
+const InstanceTypeModal: FC<InstanceTypeModalProps> = ({
+  allInstanceTypes,
+  instanceType,
+  instanceTypeVM,
+  isOpen,
+  onClose,
+  onSubmit,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const mappedInstanceTypes = useMemo(
+    () => mappedInstanceTypesToSelectOptions(allInstanceTypes),
+    [allInstanceTypes],
+  );
+  const { series: instanceTypeSeries, size: instanceTypeSize } = useMemo(
+    () => getInstanceTypeSeriesAndSize(instanceType),
+    [instanceType],
+  );
+  const [updateInProcess, setUpdateInProcess] = useState<boolean>(false);
+  const [error, setError] = useState<string>();
+
+  const [series, setSeries] = useState<string>(
+    getInstanceTypeSeriesDisplayName(mappedInstanceTypes, instanceTypeSeries),
+  );
+
+  const [size, setSize] = useState<string>(
+    getInstanceTypesPrettyDisplaySize(mappedInstanceTypes, instanceTypeSeries, instanceTypeSize),
+  );
+
+  const handleSubmit = async () => {
+    try {
+      setUpdateInProcess(true);
+      await onSubmit(
+        instanceTypeVM,
+        getInstanceTypeFromSeriesAndSize(mappedInstanceTypes, series, size),
+      );
+      onClose();
+    } catch (e) {
+      setError(e.message);
+    } finally {
+      setUpdateInProcess(false);
+    }
+  };
+
+  return (
+    <Modal
```

Comment:
Can you switch to our TabModal? this would save the actions and the handleSubmit from repeating